store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -214,7 +214,13 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         authCtxt.put(AuthContext.AC_ORIGINATING_CLIENT_IP, context.get(SoapEngine.ORIG_REQUEST_IP));
         authCtxt.put(AuthContext.AC_REMOTE_IP, context.get(SoapEngine.SOAP_REQUEST_IP));
         authCtxt.put(AuthContext.AC_ACCOUNT_NAME_PASSEDIN, acctValuePassedIn);
-        authCtxt.put(AuthContext.AC_USER_AGENT, zsc.getUserAgent());
+
+        if(zsc.getUserAgent() == null || zsc.getUserAgent() == ""){
+            authCtxt.put(AuthContext.AC_USER_AGENT, AuthContext.getDefaultUserAgent(context.get(SoapEngine.REQUEST_PROTO)));
```

Comment:
Please change this line to -
authCtxt.put(AuthContext.AC_USER_AGENT, AuthContext.getDefaultUserAgent((AuthContext.Protocol)context.get(SoapEngine.REQUEST_PROTO)));